sw/device/tests/rstmgr_sw_rst_ctrl_test.c

```diff
@@ -0,0 +1,327 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_i2c.h"
+#include "sw/device/lib/dif/dif_rstmgr.h"
+#include "sw/device/lib/dif/dif_spi_device.h"
+#include "sw/device/lib/dif/dif_spi_host.h"
+#include "sw/device/lib/dif/dif_usbdev.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/test_framework/ottf.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "i2c_regs.h"
+#include "spi_device_regs.h"
+#include "spi_host_regs.h"
+#include "usbdev_regs.h"
+
+/**
+ *  RSTMGR SW_RST_CTRL Test
+ *
+ *  This test checks RSTMGR.SW_RST_CTRL_N[index] with peripheral devices.
+ *  There are 8 RSTMGR.SW_RST_CTRL_N registers and each register
+ *  controls peripheral reset as follows:
+ *
+ * // index | device     |  test register |  reset value |  prgm value |
+ * // ------------------------------------------------------------------
+ * // 0     | SPI_DEVICE |  CFG           |  0x7f00      |  0x3f0c
+ * // 1     | SPI_HOST0  |  CONFIGOPTS    |  0x0         |  0x3210000
+ * // 2     | SPI_HOST1  |  CONFIGOPTS    |  0x0         |  0x6540000
+ * // 3     | USB        |  EP_OUT_ENABLE |  0x0         |  0xc3
+ * // 4     | USBIF      |  RXENABLE_OUT  |  0x0         |  0x695
+ * // 5     | I2C0       |  TIMING0       |  0x0         |  0x8b00cfe
+ * // 6     | I2C1       |  TIMING1       |  0x0         |  0x114010d8
+ * // 7     | I2C2       |  TIMING2       |  0x0         |  0x19ec1595
+ *
+ * 'test register' is a rw type register under each peripheral device.
+ * During the test, these registers are programmed with arbitrary values. ('prgm
+ * value') when sw resets are asserted, those values are updated to reset value.
+ * In each round, the rstmgr asserts reset to each peripheral device and
+ * each test register value is compared with the expected reset value.
+ */
+
+#define kRstMgr kTopEarlgreyResetManagerSwResetsLast + 1
```

Comment:
Probably avoid adding rstmgr to this list - it's a bit confusing / misleading.

Reply:
removed rstmgr from template